File: myblog-server/src/main/java/com/site/blog/model/dto/AjaxResultPage.java
package com.site.blog.model.dto;

import lombok.Data;
import lombok.ToString;

import java.io.Serializable;
import java.util.List;

/**分页表格数据返回
 * @author yanni
 **/
@Data
@ToString
public class AjaxResultPage<T> implements Serializable {

    /**
     * 状态码
     */
    private int code;

    /**
     * 提示消息
     */

    private String msg;

    /**
     * 总条数
     */
    private long count;

    /**
     * 表格数据
     */
    private List<T> data;


}

File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/model/dto/AjaxResultPage.kt
package com.site.blog.model.dto

import java.io.Serializable

/**分页表格数据返回
 * @author yanni
 */
data class AjaxResultPage<T>  (
    /**
     * 状态码
     */
    var  code:Int? = 0,

    /**
     * 提示消息
     */
    var  msg: String? = null,

    /**
     * 总条数
     */
    var  count: Long = 0,

    /**
     * 表格数据
     */
    var  list: List<T>? = null
)